deployment/src/main/java/io/quarkiverse/doma/deployment/DomaSettingsFactory.java

```diff
@@ -54,30 +50,9 @@ DomaSettings create() {
     }
 
     private List<DomaSettings.DataSourceSettings> dataSources() {
-        Map<String, JdbcDataSourceBuildItem> namedDataSources = dataSources.stream()
-                .collect(
-                        toMap(
-                                JdbcDataSourceBuildItem::getName,
-                                Function.identity(),
-                                (a, b) -> a,
-                                LinkedHashMap::new));
-        return namedDataSources.entrySet().stream()
-                .map(
-                        e -> {
-                            String name = e.getKey();
-                            JdbcDataSourceBuildItem dataSource = e.getValue();
-                            DomaBuildTimeConfig.DataSourceBuildTimeConfig dataSourceBuildTimeConfig;
-                            if (dataSource.isDefault()) {
-                                dataSourceBuildTimeConfig = buildTimeConfig.defaultDataSource;
-                            } else {
-                                dataSourceBuildTimeConfig = buildTimeConfig.namedDataSources.get(name);
-                                if (dataSourceBuildTimeConfig == null) {
-                                    dataSourceBuildTimeConfig = new DomaBuildTimeConfig.DataSourceBuildTimeConfig();
-                                }
-                            }
-                            return createDataSourceSettings(dataSource, dataSourceBuildTimeConfig);
-                        })
-                .collect(toList());
+        return dataSources.stream()
+                .map(ds -> createDataSourceSettings(ds, buildTimeConfig.dataSources().get(ds.getName())))
+                .toList();
```

Comment:
As I mentioned above, we can simplify things quite a lot thanks to the new features of `@ConfigMapping`.